### PR TITLE
fix: Removes update logic for the global_cluster_config resource.

### PR DESCRIPTION
**Pull request by marcosuma**
## Description

More info on https://jira.mongodb.org/browse/HELP-59447 

tl;dr: updating the global_cluster_config resource leaves the collection and index in an inconsistent state because they are on the data plane side and they would remain unchanged. 
That's also why the API does not have any PATCH endpoint.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/service/globalclusterconfig/resource_global_cluster_config_test.go [hunk 1]
```diff
@@ -116,19 +103,8 @@ func TestAccClusterRSGlobalCluster_database(t *testing.T) {
...
-					resource.TestCheckResourceAttrSet(resourceName, "custom_zone_mapping.%"),
-					resource.TestCheckResourceAttrSet(resourceName, "custom_zone_mapping.US"),
-					resource.TestCheckResourceAttrSet(resourceName, "custom_zone_mapping.IE"),
-					resource.TestCheckResourceAttrSet(resourceName, "custom_zone_mapping.DE"),
-					resource.TestCheckResourceAttrSet(resourceName, "custom_zone_mapping.JP"),
-					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
-					resource.TestCheckResourceAttr(resourceName, "cluster_name", clusterInfo.ClusterName),
-				),
+				Config:      configWithDBConfig(&clusterInfo, customZoneUpdated),
+				ExpectError: regexp.MustCompile("Updating a global cluster configuration resource is not allowed"),
```
Comment (lantoli) [inline internal/service/globalclusterconfig/resource_global_cluster_config_test.go:107]: knit: testing the update error in one test should be enough, no need to test it twice

File: website/docs/r/global_cluster_config.html.markdown [hunk 2]
```diff
@@ -13,6 +13,7 @@ description: |-
 
 -> **NOTE:** Groups and projects are synonymous terms. You may find group_id in the official documentation.
 
+~> **IMPORTANT:** A Global Cluster Configuration, once created, can only be deleted and never re-created again with the same data if not without the help of the Atlas UI. This is because the configuration and its related collection with shard key and indexes are managed separately and they would end up in an inconsistent state. [Read more about Global Cluster Configuration](https://www.mongodb.com/docs/atlas/global-clusters/)
```
Comment (lantoli) [inline website/docs/r/global_cluster_config.html.markdown]: i guess we can't prevent the resource for being created again once deleted?  it'd be nice if we could detect that conditions and block the create (the second time)

**Review by lantoli — APPROVED** [review]

**Review by oarbusi — APPROVED** [review]

File: website/docs/r/global_cluster_config.html.markdown (hunk 2 above)
Comment (maastha) [inline website/docs/r/global_cluster_config.html.markdown]: Should we include about this in the release guide too? 
Not sure if this should be considered a breaking change, I understand we are removing an incorrect behavior (just thinking out loud)

Comment (marcosuma) [reply website/docs/r/global_cluster_config.html.markdown]: yes, but I can't think of a way on our side. It should be something the Atlas itself need to implement a block at API level.

Comment (marcosuma) [reply website/docs/r/global_cluster_config.html.markdown]: It should not be considered a breaking change because it's a bug. I wasn't planning to put it in the release guide, but let me know otherwise

Comment (sarahsimpers) [inline website/docs/r/global_cluster_config.html.markdown]: Suggestion: the wording is confusing here, suggest we clarify it:
```suggestion
~> **IMPORTANT:** A Global Cluster Configuration, once created, can only be deleted. You can recreate the Global Cluster with the same data only in the Atlas UI. This is because the configuration and its related collection with shard key and indexes are managed separately and they would end up in an inconsistent state. [Read more about Global Cluster Configuration](https://www.mongodb.com/docs/atlas/global-clusters/)
```

**Review by sarahsimpers — COMMENTED** [review]
LGTM with minor suggestion

**Review by sarahsimpers — APPROVED** [review]
